src/telemetry.ts

```diff
@@ -619,6 +619,14 @@ export class IEventNamePropertyMapping {
          * Hash of the package name
          */
         hashedNamev2: string;
+        /**
+         * Whether the code is located in a Python file or a notebook cell.
+         */
+        source: 'pythonFile' | 'notebookCell';
```

Comment:
This was an ask from Python extension, to be able to differentiate between package imports in Python files vs notebooks.